README.md

```diff
@@ -0,0 +1,133 @@
+      "object":
+      {
+        "type": "coke"
+      }
+    },
+    {
+      "action": "bring",
+      "object":
+      {
+        "type": "reference"
+      },
+      "target-location":
+      {
+        "type": "person",
+        "id": "operator"
+      }
+    }
+  ]
+}
+```
+This json object can be sent to the action server.
+
+### Configuration
+
+The server parses the object and passes the resulting Python dictionary (the `recipe`) to the task manager (`set_up_state_machine(recipe)`).
+The task manager then goes through the list of actions, instantiates actions and tries to chain their semantics.
+It does this by calling the `configure` method on every action.
+
+For example, the second action (*find a coke*) results in knowledge of a coke.
+We don't know anything about this object yet, but we expect that there will be a coke in the world model.
+This knowledge is part of the `ConfigurationResult` returned by `Find.configure(configuration_data)`.
+It is added as knowledge to the configuration data that is passed to the next action.
+The next action, (*bring it to me*), receives this knowledge, but does not necessarily need it.
+However, the `reference` (*it*) in the `bring` action configuration data lets the `bring` action know that there must be knowledge of an object that can be brought somewhere.
+Therefore, it will try to find this in the knowledge in its configuration data.
+In our example, this knowledge is available, so the `bring` action will grab the found coke and take it to the operator.
+When the required knowledge is not available, the action may return a ConfigurationResult specifying that information is missing.
+The server will notify the client of this result so that it can ask the user for more information.
+
+### Execution
+
+When all actions are successfully configured, the task manager is ready to start executing the actions.
+To do this, the action server calls `task_manager.execute_next_action()` while there are still remaining actions.
+It will return early if an action fails.
+In our example: if the `find` action fails (so no coke is found), the `bring` action will not be executed.
+
+## The Action life cycle
+
+The Action life cycle consists of the following phases:
+ - Instantiation:
+    - initializing member variables (explained further in `example_action`):
+      - `_config_result`
+      - `_execute_result`
+      - `_required_field_prompts`
+      - `_required_passed_knowledge`
+      - `_required_skills`
+      - `_knowledge`
+    - the Action's implementation may add checks for static resources. E.g.
+      - knowledge availability
+      - server availability
+      - etc.
+ - Configuration
+    - checking task semantics using `_required_field_prompts`
+    - checking passed knowledge using using `_required_passed_knowledge`
+    - checking robot for required skills using `_required_skills`
+    - calling the implementation's `_configure`
+ - Execution
+    - performing the actual action
+
```

Comment:
Bit early for this level of detail, start with purpose of an Action, and how its life is related to processing the above 3 commands. Then you can also explain that actions pass information to each other. The global call interface  is useful too (construction, stuff is given to me, stuff is received from previous action, I get a ``configure`` call, I get a ``execute`` call, I may get a ``abort``. Somewhere I must produce a result and pass new information to my successor too.
At that point, I know what to expect from the outside of my action, then you can discuss its inside, with the above details.